### DISCUSSION TITLE
The best computer vision tools for environmental monitoring! 🌳

### DISCUSSION THREAD

**Discussion by NLadi** [ultralytics/ultralytics#23568] [Show and tell]
Explore the best computer vision tools for environmental monitoring that support scalable analysis of satellite imagery and environmental visual data with Abirami Vina's new blog!

Learn more ➡️ https://www.ultralytics.com/blog/the-best-computer-vision-tools-for-environmental-monitoring

[![green-woods-with-lush-vegetation-an-moss-2026-01-07-06-00-18-utc 1](https://github.com/user-attachments/assets/2d9f3dc6-ed0d-436b-80f9-35913aa8c7b9)](https://www.ultralytics.com/blog/the-best-computer-vision-tools-for-environmental-monitoring)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @NLadi, thank you for starting this discussion and for sharing the environmental monitoring blog post 🌳🚀  
This is an **automated response**, and an Ultralytics engineer will also jump in soon to help.

For anyone exploring computer vision workflows with Ultralytics, we recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples.

If this discussion is also meant to include a 🐛 Bug Report (e.g., issues when running inference/training on satellite imagery pipelines), please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) so we can help debug it efficiently.

If this is a custom training ❓ Question (for segmentation/detection on environmental datasets), please share as much detail as possible (example images, dataset format, and training logs) and confirm you’re following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) ✅

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ulalytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by glenn-jocher** [reply]
Thanks for sharing—if anyone wants to apply this to their own satellite/drone/underwater imagery, a good starting point is to tile large images into smaller patches, label a small subset, then fine-tune **Ultralytics YOLO26** for `detect` or `segment`; post your target classes plus a couple sample images (and dataset format) in this thread and we can help you pick sensible training settings (see the blog: [The best computer vision tools for environmental monitoring](https://www.ultralytics.com/blog/the-best-computer-vision-tools-for-environmental-monitoring) and the [YOLO26 model docs](https://docs.ultralytics.com/models/yolo26/)).
